### PR TITLE
Add the gRPC-based envelope spec.

### PR DESCRIPTION
**Pull request by danilvpetrov**
#### What change does this introduce?

This change  relocates envelope spec definitions from the github.com/dogmatiq/envelopespec repository.

#### What issues does this relate to?

Partially addresses https://github.com/dogmatiq/interopspec/issues/1.

#### Why make this change?

This change is required to relocated the envelope spec into a dedicated package within the current repository. This change is among others to arrange and concentrate all spec definitions for application interoperability within a single repository.

#### Is there anything you are unsure about?

Possibly the directory structure for the envelope spec within the current repository might be reviewed.

### CODE REVIEW

File: envelopespec/envelope.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package envelopespec
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Validate returns an error if env is not well-formed.
```
Comment (danilvpetrov) [inline envelopespec/envelope.go:8]: I am hesitant to remove this section of doc comments just to replace it with `Validate returns an error if m contains invalid fields.`. The existing doc comments seem to cover all the peculiarities fo the envelope validation.

Comment (jmalloc) [reply envelopespec/envelope.go:8]: Yep, it can stay as is.

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,11 @@ The format is based on [Keep a Changelog], and this project adheres to
 
 ## [Unreleased]
 
+### Added
+
+- Added the content of 'github.com/dogmatiq/envelopespec' repository into
+  `envelope` subdirectory.
```
Comment (jmalloc) [inline CHANGELOG.md:17]:
```suggestion
- Add `envelopespec` package
```

I don't think there's any value to referring to this old repo which will be removed.

File: envelopespec/envelope_test.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+import (
+	"time"
+
+	. "github.com/dogmatiq/interopspec/envelopespec"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("func CheckWellFormed()", func() {
```
Comment (jmalloc) [inline envelopespec/envelope_test.go:11]:
```suggestion
var _ = Describe("type Envelope", func() {
	Describe("func Validate()", func() {
```

**Review by jmalloc — CHANGES_REQUESTED** [review]

**Review by jmalloc — APPROVED** [review]

**Pull request merged**